lib/viewRenderer/nodeRendering/ShapeBuilder.ts

```diff
@@ -0,0 +1,339 @@
+      withDashedStroke,
+      textAnchor,
+      refX,
+    });
+
+    attributes.body.rx = 8;
+    attributes.body.ry = 8;
+
+    element.resize(width, height);
+    element.attr(attributes as unknown as dia.Cell.Selectors);
+
+    return element;
+  }
+
+  /**
+   * Builds a basic octagonal shape
+   * @param elementName Element name
+   * @param settings Shape settings
+   * @param settings.width Reference size for element's width
+   * @param settings.height Reference size for element's height
+   * @param settings.textAnchor Text Anchor
+   * @param settings.refX Label x-axis reference
+   * @param settings.fillColor Color used to fulfill relationship arrow shapes
+   * @param settings.strokeColor Color used for strokes
+   * @param settings.withDashedStroke Enable or disable Dashed Stroke
+   * @return Basic octagonal shape
+   * @example
+   *  import { ShapeBuilder } from '@lib/viewRenderer/nodeRendering/ShapeBuilder';
+   *  import { ViewSettings } from '@lib/viewRenderer/ViewSettings';
+   *
+   *  const shapeBuilder = new ShapeBuilder(new ViewSettings({}));
+   *  const node = shapeBuilder.buildBasicOctagonal('Node 1',{
+   *       strokeColor: 'black',
+   *       fillColor: 'white',
+   *       refX: '42.5%',
+   *       textAnchor: 'middle',
+   *       width: 100,
+   *       height: 80
+   * });
+   */
+  buildBasicOctagonal(
+    elementName,
+    {
+      width = this.defaultWidth,
+      height = this.defaultHeight,
+      textAnchor = 'middle',
+      refX = '42.5%',
+      fillColor = this.lightColor,
+      strokeColor = this.darkColor,
+      withDashedStroke = false,
+    }: Settings,
+  ) {
+    const element = new shapes.standard.Polygon();
+    const attributes = this.getAttributes({
+      elementName,
+      fillColor: fillColor || this.lightColor,
+      strokeColor: strokeColor || this.darkColor,
+      withDashedStroke,
+      textAnchor,
+      refX,
+    });
+
+    attributes.body.refPoints = '0,1 0.5,0 9.5,0 10,1 10,9 9.5,10 0.5,10 0,9';
+
+    element.resize(width, height);
+    element.attr(attributes as unknown as dia.Cell.Selectors);
+
+    return element;
+  }
+
+  /**
+   * Builds a small circle shape
+   * @param settings Shape settings
+   * @param settings.fillColor Color used to fulfill relationship arrow shapes
+   * @param settings.strokeColor Color used for strokes
+   * @return Basic small circle shape
+   * @example
+   *  import { ShapeBuilder } from '@lib/viewRenderer/nodeRendering/ShapeBuilder';
+   *  import { ViewSettings } from '@lib/viewRenderer/ViewSettings';
+   *
+   *  const shapeBuilder = new ShapeBuilder(new ViewSettings({}));
+   *  const node = shapeBuilder.buildSmallCircle({
+   *       strokeColor: 'black',
+   *       fillColor: 'white',
+   * });
+   */
+  buildSmallCircle({
+    fillColor = this.darkColor,
+    strokeColor = this.darkColor,
+  }: Partial<Omit<ShapeAttributes, 'elementName' | 'withDashedStroke' | 'textAnchor' | 'refX'>>) {
```

Comment:
Do you think it would be interesting to extract a new type and use it here?

`type ShapeColorOption = Partial<Omit<ShapeAttributes, 'elementName' | 'withDashedStroke' | 'textAnchor' | 'refX'>>`

Reply:
Yes, I believe it would be interesting to extract a new type. I will fix it.

Reply:
I fixed it